polly/plugins/maya/publish/submit_deadline.py

```diff
@@ -25,7 +25,8 @@ def process(self, instance):
         from avalon import api
         from avalon.vendor import requests
 
-        assert api.Session["AVALON_DEADLINE"], "Requires AVALON_DEADLINE"
+        deadline = api.Session.get("AVALON_DEADLINE", None)
```

Comment:
No, I think I'd prefer to handle dependencies before having it move on. It simplifies debugging and readability.